nautobot/core/api/serializers.py

```diff
@@ -119,6 +136,34 @@ def __init__(self, *args, **kwargs):
                 kwargs["view_name"] = get_route_for_model(kwargs["queryset"].model, "detail", api=True)
         super().__init__(*args, **kwargs)
 
+    def to_internal_value(self, data):
+        """Convert potentially nested representation to a model instance."""
+        if isinstance(data, dict):
+            if "url" in data:
+                return super().to_internal_value(data["url"])
+            elif "id" in data:
+                return super().to_internal_value(data["id"])
+        return super().to_internal_value(data)
+
+    def to_representation(self, value):
+        """Convert URL representation to a brief nested representation."""
+        url = super().to_representation(value)
+        if self.queryset:
+            model = self.queryset.model
+        elif isinstance(self.parent, ManyRelatedField) and getattr(self.parent.parent.Meta.model, self.source, False):
+            model = getattr(self.parent.parent.Meta.model, self.source).field.model
+        elif getattr(self.parent.Meta.model, self.source, False):
+            model = getattr(self.parent.Meta.model, self.source).field.model
+        else:
+            logger.warning(
+                "Unable to determine model for related field %r; "
+                "ensure that either 'queryset' is set on the field "
+                "or defines the related model in the Meta.",
+                self.field_name,
+            )
+            return {"id": value.pk, "object_type": "unknown", "url": url}
```

Comment:
Perhaps:

```suggestion
            return {"id": value.pk, "object_type": "unknown.unknown", "url": url}
```

Just so we stick to the `app_label.modelname` structure?